### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by zpeganov**
Solo commit due to only sitting in for lecture

### CODE REVIEW

**Comment by Al3xramirez** [comment]
Great job! I love your new functionalities you added like the "hint"

One little thing I would consider is possibly revising your DisplayHint_ReturnsStringAnswerIndex name into something like DisplayHint_ReturnsCorrectAnswerIndexString since this is a very quick way to understand the intent. of the test A very nit picky thing I know 😂

You may also want to consider some invalid input handing and edge cases when the trivia file is empty. Also,  I think it would be beneficial to the user if you had the DisplayHint returning the actual answer text instead of the number.

Pull Request title includes "Assignment 1" in the title ✔
Partner has also made a commit ❌
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Extra Credit
Come up with a new feature for the application
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 1]
```diff
@@ -1,37 +1,57 @@
...
+           
         return DisplayResult(userGuess, question);
     }
 
+    public static string DisplayHint(Question question)
+    {
+        return "The correct answer is: " + question.CorrectAnswerIndex;
+    }
+
     public static string GetGuessFromUser()
```
Comment (RyanHirte) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:49]: Something that was suggested to me in my review was validating the user input. If I input the correct answer by typing the string, it will tell me it's incorrect. Validating that the input is either "hint" or a proper number aka 1-3 could be a way to make this program more robust.

**Review by RyanHirte — COMMENTED** [review]
Pull Request title includes "Assignment 1" in the title ✔
Partner has also made a commit ❌
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Extra Credit
Come up with a new feature for the application
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

Overall great job and I like the hint feature. Along with deleting commented lines to clean things up and my suggestion on input validation I think this is good!

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
 namespace PrincessBrideTrivia.Tests;
 
 [TestClass]
 public class ProgramTests
 {
+
+    [TestMethod]
+    public void DisplayHint_ReturnsStringAnswerIndex()
+    {
+        //Arrange
```
Comment (BraydenMcMahon) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:12]: Since the code looks solid just want to nit pick on formatting and the 3 lines of comments dont seem to clarify much as to what the following line is does.

**Review by BraydenMcMahon — COMMENTED** [review]
Pull Request title includes "Assignment 1" in the title ✔
Partner has also made a commit ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔

**Comment by kellanburns** [comment]
Pull Request title includes "Assignment 1" in the title ✔
Partner has also made a commit ❌
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Extra Credit
Come up with a new feature for the application
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

As everyone else seems to be noticing, you did some great work man. It looks like AI/ autocomplete suggestions may have been fighting you, because some of these funny things like unused namespace imports have been popping up in my code more recently too. Not an issue, but could be worth removing for a final product.

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 3]
```diff
@@ -1,37 +1,57 @@
...
 public class Program
 {
     public static void Main(string[] args)
     {
         string filePath = GetFilePath();
         Question[] questions = LoadQuestions(filePath);
-
+        Console.WriteLine("Type HINT for a hint");
         int numberCorrect = 0;
+        // Console.WriteLine(questions.Length);
```
[the remark this answers is not in the document]
Comment (Hellothereyoko) [reply PrincessBrideTrivia/PrincessBrideTrivia/Program.cs]: Good catch! I don't think I caught that

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 4]
```diff
@@ -1,8 +1,25 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (Joshua-Lester3) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:1]: nit: I think this is an unused using directive.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 5]
```diff
@@ -1,8 +1,25 @@
...
+
 namespace PrincessBrideTrivia.Tests;
 
 [TestClass]
 public class ProgramTests
 {
+
+    [TestMethod]
+    public void DisplayHint_ReturnsStringAnswerIndex()
```
Comment (Joshua-Lester3) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs]: Test method names should follow `MethodUnderTest_ConditionUnderTest_ExpectedResult`. Though the base assignment1 branch doesn't - I submitted a PR to fix this.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs (hunk 2 above)
Comment (Joshua-Lester3) [reply PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:12]: @BraydenMcMahon these are to organize test methods :) I think Ben might've went over them last Tuesday, but I'm not sure because I wasn't there.

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 6]
```diff
@@ -1,37 +1,57 @@
...
         Console.WriteLine("You got " + GetPercentCorrect(numberCorrect, questions.Length) + " correct");
     }
 
     public static string GetPercentCorrect(int numberCorrectAnswers, int numberOfQuestions)
     {
-        return (numberCorrectAnswers / numberOfQuestions * 100) + "%";
+        // Console.WriteLine("Number correct: " + numberCorrectAnswers);
+        // Console.WriteLine("Number of questions: " + numberOfQuestions);
+        // Console.WriteLine(((double)numberCorrectAnswers / (double)numberOfQuestions));
+        double result = (double)numberCorrectAnswers / (double)numberOfQuestions * 100;
```
Comment (Joshua-Lester3) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs]: Only one of the (double) casts is necessary here.

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 7]
```diff
@@ -85,6 +107,8 @@ public static Question[] LoadQuestions(string filePath)
             question.Answers[0] = answer1;
             question.Answers[1] = answer2;
             question.Answers[2] = answer3;
+            questions[i] = question;
+
             question.CorrectAnswerIndex = correctAnswerIndex;
```
[the remark this answers is not in the document]
Comment (Joshua-Lester3) [reply PrincessBrideTrivia/PrincessBrideTrivia/Program.cs]: This is incorrect - since Question is a reference type, it'll update the reference to the specific object :)

**Review by Joshua-Lester3 — COMMENTED** [review]
## Assignment Details

* Pull Request title includes "Assignment 1" in the title ✔
* Partner has also made a commit ✔❌ not applicable here
* Pull Request targets Assignment 1 branch ✔
* Issue 1: Application no longer crashes ✔
* Issue 2: Unit test properly passes ✔

## Extra Credit

* Come up with a new feature for the application
  * Issue 3: Add a new feature request implemented ✔
  * Issue 3: Feature request unit tested ✔

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 8]
```diff
@@ -56,6 +76,8 @@ public static void DisplayQuestion(Question question)
         {
             Console.WriteLine((i + 1) + ": " + question.Answers[i]);
         }
+         
+
```
[the remark this answers is not in the document]
Comment (zpeganov) [reply PrincessBrideTrivia/PrincessBrideTrivia/Program.cs]: Done
